micro-ROS/micro_ros_espidf_component
# Added Int32 subscriber to publisher example.

As discussed in #8, I finally found time to create the pull request.

To test the subscriber:
1. Build and flash the code as per the instructions on the README.md using the directory `examples/int32_sub_pub`.
1. Monitor the output of the ESP32 using `idf.py monitor`.  You should see `Sent: ` with an incrementing number every second.
1. In a new shell, run the agent docker using:
    ```bash
    docker run -it --rm --net=host microros/micro-ros-agent:foxy udp4 --port 8888 -v6
    ```
1. The output from the agent should now show messages being passed to and from the agent. 
1. In another shell, find the value of the container ID using `docker ps`
1. Attach the shell to the agent docker using: 
    ```bash
    docker exec -it  <container ID> /bin/bash
    ```
1. Verify that the publisher can be seen by ROS2.
    ```bash
    # ros2 topic list
    /int32_publisher
    /int32_subscriber
    /parameter_events
    /rosout
    # ros2 topic echo /int32_publisher
    data: 1589
    ---
    data: 1590
    ---
    ```
1. Now get the info on the subscriber and publish to it.
    ```bash
    # ros2 topic info /int32_subscriber
    Type: std_msgs/msg/Int32  
    Publisher count: 0
    Subscription count: 2
    # ros2 topic pub /int32_subscriber std_msgs/msg/Int32 "{data: 345}"
    ```
1.  Once the publisher is running, switch back to the shell monitoring the ESP32.  You should see a message being printed out every second showing something like this:
    ```bash
    Received: 345
    Sent: 2078
    Received: 345
    Sent: 2079
    ```

That's it! A subscriber and a publisher working at the same time.     

Signed-off-by: Andy Blight <andy.blight@gmail.com>
